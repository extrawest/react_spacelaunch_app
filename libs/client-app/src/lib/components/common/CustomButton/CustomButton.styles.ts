import type { SxProps, Theme } from '@mui/material';

const button: SxProps<Theme> = (theme) => ({
  px: 10,
  py: 3.75,
  borderRadius: 10,
  background: `linear-gradient(93.72deg, ${theme.palette.secondary.main} 9.41%, ${theme.palette.primary.main} 86.1%)`,
  boxShadow: 0,
});

export const styles = {
  button,
};
