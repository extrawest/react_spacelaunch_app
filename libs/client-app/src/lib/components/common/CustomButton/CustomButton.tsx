import type { FC } from 'react';

import { Button as MuiButton } from '@mui/material';

import { styles } from './CustomButton.styles';
import type { CustomButtonProps } from './CustomButton.types';

export const CustomButton: FC<CustomButtonProps> = ({ children }) => {
  return (
    <MuiButton
      variant="contained"
      sx={styles.button}
    >
      {children}
    </MuiButton>
  );
};
