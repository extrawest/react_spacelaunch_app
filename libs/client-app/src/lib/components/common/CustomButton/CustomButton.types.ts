export interface CustomButtonProps {
  children: string;
}
