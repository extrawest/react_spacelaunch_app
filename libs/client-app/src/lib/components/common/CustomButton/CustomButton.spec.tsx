import { fireEvent, render } from '@testing-library/react';

import { CustomButton } from './CustomButton';

describe('[component]: CustomButton', () => {
  const text = 'Text';
  const mockFn = jest.fn();

  test('Should match default style', () => {
    const { baseElement } = render(
      <CustomButton onClick={mockFn}>{text}</CustomButton>
    );

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Text should be correct', () => {
    const { getByText } = render(
      <CustomButton onClick={mockFn}>{text}</CustomButton>
    );

    expect(getByText(text)).toHaveTextContent(text);
  });

  test('Click event should work', () => {
    const { getByRole } = render(
      <CustomButton onClick={mockFn}>{text}</CustomButton>
    );

    fireEvent.click(getByRole('button'));
    expect(mockFn).toBeCalled();
  });

  afterEach(() => {
    jest.clearAllMocks();
    mockFn.mockClear();
  });
});
