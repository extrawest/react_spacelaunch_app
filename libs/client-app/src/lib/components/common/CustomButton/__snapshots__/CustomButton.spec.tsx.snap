// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]: CustomButton Should match default style 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-aal01w-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Text
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
